Userland/Libraries/LibCore/ArgsParser.cpp

```diff
@@ -430,6 +430,23 @@ void ArgsParser::add_option(DeprecatedString& value, char const* help_string, ch
     add_option(move(option));
 }
 
+void ArgsParser::add_option(String& value, char const* help_string, char const* long_name, char short_name, char const* value_name, OptionHideMode hide_mode)
+{
+    Option option {
+        OptionArgumentMode::Required,
+        help_string,
+        long_name,
+        short_name,
+        value_name,
+        [&value](StringView s) {
+            value = String::from_utf8(s).release_value_but_fixme_should_propagate_errors();
+            return true;
```

Comment:
Yes and yes! And it wouldn't be that major of a change; you'll just have to update all the implementors, and have `ArgsParser` print the error string instead of just "Invalid value for option" as it is now.

And you'd want to differentiate between errors due to bad input (such as invalid UTF-8 here -- which you want to report to the user as an invalid option value) and generic errors that prevented you from accepting the value (such as `ENOMEM` -- you probably want to either propagate these to the caller). Maybe `is_errno()` is sufficient for this, maybe not.

Reply:
Cool, I'll make the simpler change at some point today (hopefully) and then see if I can raise another PR for the wider ArgsParser change :)

Reply:
@bugaevc @BenWiederhake Apologies for the long lead time on this, good old life happened so I have only had a chance to take a look at this today.

I believe my latest fixup to the argsparser commit resolves this comment. Please let me know if it does not.